File: 0868-binary-gap/0868-binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        max_gap = curr = 0
        prev = -1
        while(n):
            if(n & 1):
                if(prev != -1):
                    max_gap = max(curr - prev, max_gap)
                prev = curr
                
            n >>= 1
            curr += 1
        
        return max_gap

File: 0868-binary-gap/0868-binary-gap.go
func binaryGap(n int) int {
    prev, curr := -1, -1
    res, i := 0, 0
    for n > 0 {
        if (n & 1) == 1 {
            prev = curr
            curr = i
            if prev != -1 {
                res = max(res, curr - prev)
            }
        }
        i++
        n >>= 1
    }
    return res
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}